lib/buffer.js

```diff
@@ -462,8 +462,11 @@ Buffer.prototype.toString = function() {
   } else {
     result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
+  if (result === undefined) {
+    if (this.length > binding.kStringMaxLength)
+      throw new Error('Size exceeds max string length');
```

Comment:
we treat all error message changes as semver-major. Hopefully I'll be able to finish up the externalized messages support soon so we can move away from that. In the meantime, +1 for this being a RangeError